src/components/Calculator.jsx

```diff
@@ -36,11 +38,54 @@ export const Calculator = () => {
         result: "Enter a valid number",
       });
 
-    return setData({ ...data, [v.target.name]: v.target.value });
+    const newData = { ...data, [v.target.name]: v.target.value };
+    const res = operations.operation(
+      parseFloat(newData.a),
+      parseFloat(newData.b),
+      newData.operation
+    );
+
+    const newHistory = [...history.slice(0, historyPos), newData];
```

Comment:
Why are you using slice instead of just destructure the history and add the new element?

Reply:
of course it changed that way